src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -35,6 +34,7 @@ export class ComplexResultsComponent implements OnInit, AfterViewInit {
 
   ngOnInit() {
     this.titleService.setTitle('Complex Portal - Results');
+    this._allComponentsInComplexSearch = new Set();
```

Comment:
I noticed some errors when I search, then went to the home page and search again. I think it was related to `_allComponentsInComplexSearch` not being reset to an empty set, so that is handled here now.